PathPlanning/RRTStar/rrt_star.py

```diff
@@ -58,28 +60,35 @@ def planning(self, animation=True, search_until_max_iter=True):
         """
         rrt star path planning
 
-        animation: flag for animation on or off
-        search_until_max_iter: search until max iteration for path improving or not
+        animation: flag for animation on or off search_until_max_iter: search
+        until max iteration for path improving or not
         """
 
         self.node_list = [self.start]
         for i in range(self.max_iter):
             print("Iter:", i, ", number of nodes:", len(self.node_list))
             rnd = self.get_random_node()
             nearest_ind = self.get_nearest_node_index(self.node_list, rnd)
-            new_node = self.steer(self.node_list[nearest_ind], rnd, self.expand_dis)
+            new_node = self.steer(
+                self.node_list[nearest_ind], rnd, self.expand_dis)
+            near_node = self.node_list[nearest_ind]
+            new_node.cost = near_node.cost + \
+                math.sqrt((new_node.x-near_node.x)**2 +
+                          (new_node.y-near_node.y)**2)
 
             if self.check_collision(new_node, self.obstacle_list):
                 near_inds = self.find_near_nodes(new_node)
-                new_node = self.choose_parent(new_node, near_inds)
-                if new_node:
+                new_node_2 = self.choose_parent(new_node, near_inds)
+                if new_node_2:
+                    self.rewire(new_node_2, near_inds)
+                    self.node_list.append(new_node_2)
+                else:
```

Comment:
This is fixing for #382, correct? If new_node_2 is None, that will be added node_list. I agree to fix it. However, I think `new_node_2` should be `parent_updated_node` or something is better.

Reply:
`new_node_2` is now `node_with_updated_parent`.

Reply:
Yes, this should fix #382